src/Microsoft.Maintainability.Analyzers/Core/DoNotIgnoreMethodResults.cs

```diff
@@ -1,68 +1,146 @@
+                                                                             "{0}",     // Use a placeholder message format as we need to display different messages based on the violation.
                                                                              DiagnosticCategory.Performance,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182273.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(ObjectCreationRule, StringCreationRule, HResultOrErrorCodeRule, TryParseRule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterOperationBlockStartAction(osContext =>
+            {
+                var method = osContext.OwningSymbol as IMethodSymbol;
+                if (method == null)
+                {
+                    return;
+                }
+
+                osContext.RegisterOperationAction(opContext =>
+                {
+                    var expression = ((IExpressionStatement)opContext.Operation).Expression;
+                    string messageFormat = null;
+                    string targetMethodName = null;
+                    switch (expression.Kind)
+                    {
+                        case OperationKind.ObjectCreationExpression:
+                            var ctor = ((IObjectCreationExpression)expression).Constructor;
+                            if (ctor != null)
+                            {
+                                messageFormat = MicrosoftMaintainabilityAnalyzersResources.DoNotIgnoreMethodResultsMessageObjectCreation;
+                                targetMethodName = ctor.ContainingType.Name;
+                            }
+                            break;
+
+                        case OperationKind.InvocationExpression:
+                            var invocationExpression = ((IInvocationExpression)expression);
+                            var targetMethod = invocationExpression.TargetMethod;
+                            if (targetMethod == null)
```

Comment:
I don't think it is possible for semantically well-formed code. I don't know if there are cases where one of the compilers can generate a bound invocation with a null target in the presence of errors.